spring-cloud-gateway-integration-tests/grpc/src/main/java/org/springframework/cloud/gateway/tests/grpc/GRPCApplication.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.tests.grpc;
+
+import io.grpc.Grpc;
+import io.grpc.Server;
+import io.grpc.ServerCredentials;
+import io.grpc.TlsServerCredentials;
+import io.grpc.stub.StreamObserver;
+
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Alberto C. Ríos
+ */
+@SpringBootConfiguration
+@EnableAutoConfiguration
+public class GRPCApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(GRPCApplication.class, args);
+	}
+
+	@Bean
+	public RouteLocator customRouteLocator(RouteLocatorBuilder builder) {
+		return builder.routes()
+				.route("grpc", r -> r.predicate(p -> true).uri("https://localhost:50052"))
+				.build();
+	}
+
+	@Component
+	static class GRPCServer implements ApplicationRunner {
+
+		private Server server;
+
+		@Override
+		public void run(ApplicationArguments args) throws Exception {
+			final GRPCServer server = new GRPCServer();
+			server.start();
+		}
+
+		private void start() throws Exception {
+			/* The port on which the server should run */
+			int port = 50052;
+			ServerCredentials creds = createServerCredentials();
+			server = Grpc.newServerBuilderForPort(port, creds)
+					.addService(new HelloService()).build().start();
+
+			System.out.println("Starting server in port " + port);
+
+			Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+				try {
+					GRPCServer.this.stop();
+				}
+				catch (InterruptedException e) {
+					e.printStackTrace(System.err);
+				}
+			}));
+		}
+
+		private ServerCredentials createServerCredentials() throws IOException {
+			File privateKey = new ClassPathResource("private.key").getFile();
```

Comment:
I prefer testing with a real private key and cert chain in the server, but let me know if you think this is too much and I should use something else.